### PR TITLE
SE-1321 add multiple files upload allowed option [premerges]

### PR DESCRIPTION
**Pull request by jansenk**
**TL;DR -**  This is the content of https://github.com/edx/edx-ora2/pull/1256 with master merged in, conflicts resolved, and changes made for Team ORAs

**What changed?**

- Added an option for allowing/disallowing multiple files to be uploaded for an assignment. Since multiple files is the default, what this really does is provide a mechanism for instructors to restrict responses to a single uploaded file. 
- For Team assignments, after talking with @sapanathomas523, we decided it made the most sense to ignore this setting for team assignments. If an ora is set to teams, we lock this value to True.

**Developer Checklist**
- [x] Reviewed the [release process](./release_process.md)
- [x] Translations up to date
- [x] JS minified, SASS compiled
- [x] Test suites passing on Jenkins
- [x] Bumped version number in [setup.py](../setup.py) and [package.json](../package.json)
- [x] Wording signed off by @sapanathomas523 

JIRA: [OSPR-3768](https://openedx.atlassian.net/browse/OSPR-3768)

FIY: @edx/masters-devs-gta

***Original PR Body***

This PR adds an option to toggle between allowing multiple files uploaded and only allowing a single file. This is fully backwards compatible. Currently behaviour is to allow students to select an arbitrary number of files for the file upload response. Adding this option allows course authors to limit this to a single file upload.

For now, this is to be treated as a prototype. Hence no tests or updated translations yet.

**JIRA tickets**: [OSPR-3768](https://openedx.atlassian.net/browse/OSPR-3768)

**Dependencies**: None

**Screenshots**: 

**Testing instructions**:

1. create a new ora assessment on a course in studio
2. in the settings, make sure the file response is required or optional
3. below that in the settings, change "allow multiple files" to false (it must be true by default)
4. save and publish the ora
5. open the ora in the lms
6. click the button to upload files and confirm that only one file can be selected (ctl+click on multiple files will select multiple files if enabled)
7. select a single file and upload it. verify that it uploads successfully
8. repeat uploading a file to verify that uploading a new file, replacing the old file, still works
9. using the browser dev tools, inspect the upload file input and add the `multiple` attribute.
10. attempt to upload multiple files by selecting multiple files, adding a description for each one, and clicking upload files
11. verify that this is rejected server-side with an error message
12. refresh the page and verify that only the single, previously successfully uploaded file is visible
13. create a new ora assessment with a file response, but leaving "allow multiple files" as True
14. attempt to upload multiple files on this new assessment and verify that this still works successfully

### CODE REVIEW

**Comment by jansenk** [comment]
Screen Shot 2020-10-14 at 12 16 55 PM
Screen Shot 2020-10-14 at 12 16 39 PM

File: openassessment/xblock/submission_mixin.py [hunk 1]
```diff
@@ -414,6 +414,20 @@ def upload_url(self, data, suffix=''):  # pylint: disable=unused-argument
...
+
+            # Here we check if there are existing file uploads by checking for
+            # an existing download url for any of the upload slots.
+            # Note that we can't use self.saved_files_descriptions because that
+            # is populated before files are uploaded
+            for i in range(self.MAX_FILES_COUNT):
+                file_url = self._get_download_url(i)
+                if file_url:
+                    return {'success': False,
+                            'msg': self._(u"Only a single file upload is allowed for this assessment.")}
```
Comment (Andytr1) [inline openassessment/xblock/submission_mixin.py:427]: nit directed at original author: we no longer need to use 'u' prefix.

**Review by Andytr1 — APPROVED** [review]
Code looks good overall.

File: openassessment/xblock/submission_mixin.py (hunk 1 above)
Comment (jansenk) [reply openassessment/xblock/submission_mixin.py:427]: back when he wrote this, we did 😆

**Review by nsprenkle — APPROVED** [review]
Tested locally + code review LGTM :shipit:
